src/txn/txn_recover.c

```diff
@@ -732,6 +732,16 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
         goto done;
     }
 
+    if (!hs_exists) {
+        __wt_verbose(session, WT_VERB_RECOVERY | WT_VERB_RECOVERY_PROGRESS, "%s",
+          "Creating the history store before applying log records. Likely recovering after an"
+          "unclean shutdown on an earlier version");
```

Comment:
Is there any way that doing this like this can leave the database in a bad state? As in the configuration is wrong and messes something up later or if the user then downgrades again?

@agorrod in the ticket you mentioned (and I agree) that the correct place about this is eviction. It pre-emptively opens a cursor to the HS. Can it do that only if it exists? That feels like the right place. This feels kind of random in recovery and is really unrelated to recovery itself. If it doesn't exist, eviction should never need it.

Reply:
I replied in the JIRA ticket.